spec/socialprovider.spec.js

```diff
@@ -102,6 +102,19 @@ describe("Tests for message batching in Social provider", function() {
     expect(xmppSocialProvider.client.send).toHaveBeenCalled();
   });
 
+  it("sends message to correct destinations", function() {
+    xmppSocialProvider.sendMessage('Bob', 'Hi', function() {});
+    xmppSocialProvider.sendMessage('Alice', 'Hi', function() {});
+    expect(xmppSocialProvider.client.send).not.toHaveBeenCalled();
+    jasmine.clock().tick(100);
+    expect(xmppSocialProvider.client.send.calls.count()).toEqual(2);
+    var dest = xmppSocialProvider.client.send.calls.first().args[0].parent.attrs.to;
+    if (dest !== 'Bob' && dest !== 'Alice') {
+      expect(dest).toEqual('Bob');
```

Comment:
I'm a little confused by this test case.  Won't this ```expect(dest).toEqual('Bob');``` fail, since this statement is only hit if dest !== 'Bob' as per the above if statement?